### PR TITLE
Snap shot Test Cases (new branch continue from #74)

### PR DESCRIPTION
**Pull request by LiuYuHao**
Totally 5 ~ 6 hours (spent another hour on fix and git hub issues(rebase problem, I am new to it).

### CODE REVIEW

File: website/package.json [hunk 1]
```diff
@@ -44,6 +44,7 @@
     "eslint-plugin-import": "2.14.0",
     "eslint-plugin-jsx-a11y": "6.1.2",
     "eslint-plugin-react": "7.11.1",
+    "firebase-mock": "2.2.10",
```
Comment (khelif96) [inline website/package.json]: We arent using this remove it

File: website/src/components/EventSummary/EventSummary.test.jsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import {
+  configure,
+} from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import { EventSummary } from '..';
+
+configure({ adapter: new Adapter() });
+
+describe('EventSummary', () => {
+  it('Render EventSummary without any eror', () => { // eslint-disable-line no-undef
```
Comment (khelif96) [inline website/src/components/EventSummary/EventSummary.test.jsx]: You can remove these comments i resolved it in the eslint config

File: website/src/components/EventSummaryCollection/EventSummaryCollection.test.jsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    {
+      course: 'CSC 220',
+      eventName: 'Homework 1',
+      dueTime: '5:00pm',
+      color: '#62c4f9',
+    },
+  ],
+};
+
+// problem need unique key for each iteration
```
Comment (khelif96) [inline website/src/components/EventSummaryCollection/EventSummaryCollection.test.jsx]: add a key value to each object

File: website/src/components/EventSummaryCollection/EventSummaryCollection.test.jsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+describe('EventSummaryCollection', () => {
+  it('Render EventSummaryCollection without any eror', () => { // eslint-disable-line no-undef
```
Comment (khelif96) [inline website/src/components/EventSummaryCollection/EventSummaryCollection.test.jsx]: Remove the comments

File: website/src/components/HeaderIcons/HeaderIcons.test.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  configure,
+} from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import { MemoryRouter } from 'react-router-dom';
+import { HeaderIcons } from '..';
+
+configure({ adapter: new Adapter() });
+
+describe('Headerlcons', () => {
+  it('Render Headerlcons without any eror', () => { // eslint-disable-line no-undef
```
Comment (khelif96) [inline website/src/components/HeaderIcons/HeaderIcons.test.jsx]: Comments

File: website/src/components/IconButton/IconButton.test.jsx [hunk 6]
```diff
@@ -1,24 +1,26 @@
 import React from 'react';
 import renderer from 'react-test-renderer';
-import { shallow, mount, render, configure} from 'enzyme';
+import {
+  shallow,
```
Comment (khelif96) [inline website/src/components/IconButton/IconButton.test.jsx]: Maybe write more tests to try out different types of icons and see if they render properly

File: website/src/components/MaterialIcon/MaterialIcon.test.jsx [hunk 7]
```diff
@@ -1,14 +1,16 @@
...
 import Adapter from 'enzyme-adapter-react-16';
-import { Button, Card, Icon } from 'antd';
 import { MaterialIcon } from '..';
 import '../../utils/tests/test.css';
 
 configure({ adapter: new Adapter() });
 
-describe("MaterialIcon", () => {
+describe('MaterialIcon', () => {
   it('Expect MaterialIcon to return an icon with a settings cog', () => { // eslint-disable-line no-undef
```
Comment (khelif96) [inline website/src/components/MaterialIcon/MaterialIcon.test.jsx]: Comments in this file

File: website/src/views/CreateClass/CreateClass.jsx [hunk 8]
```diff
@@ -1,30 +1,30 @@
-import React, { PureComponent } from 'react';
-import firebase from 'firebase';
+import React, { Component } from 'react';
```
Comment (khelif96) [inline website/src/views/CreateClass/CreateClass.jsx]: Was this file supposed to be changed? I thought this was just testing

File: website/src/views/CreateClass/CreateClass.test.jsx [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+import {
+  configure,
+} from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import { CreateClass } from '..';
+
+configure({ adapter: new Adapter() });
+
+describe('CreateClass', () => {
+  it('Render CreateClass without any eror', () => { // eslint-disable-line no-undef
```
Comment (khelif96) [inline website/src/views/CreateClass/CreateClass.test.jsx]: Comments

File: website/src/views/DashboardHome/DashboardHome.test.jsx [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+  configure,
+} from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import { DashboardHome } from '..';
+
+configure({ adapter: new Adapter() });
+
+
+describe('DashboardHome', () => {
+  it('Render DashboardHome without any eror', () => { // eslint-disable-line no-undef
```
Comment (khelif96) [inline website/src/views/DashboardHome/DashboardHome.test.jsx]: Comments

File: website/src/views/Signin/Signintest.jsx [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+        email: 'test@test.com',
+        emailVerified: true,
+        getIdToken: () => new Promise(((resolve) => {
+          setTimeout(() => {
+            resolve(1234);
+          }, 0);
+        })),
+      },
+    }));
+  it('Render Signin without any eror', () => { // eslint-disable-line no-undef
```
Comment (khelif96) [inline website/src/views/Signin/Signintest.jsx]: Comments

File: website/src/views/Signin/Signintest.jsx [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+  configure,
+} from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import firebase from 'firebase';
+import { Signin } from '..';
+
+configure({ adapter: new Adapter() });
+
+describe('Signin', () => {
+  jest.spyOn(firebase, 'auth')
```
Comment (khelif96) [inline website/src/views/Signin/Signintest.jsx]: You can also mock the fetch function to make this test work

File: website/src/views/Signup/Signuptest.jsx [hunk 13]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
```
Comment (khelif96) [inline website/src/views/Signup/Signuptest.jsx]: Same comments from signintest.jsx

File: website/src/views/Signup/Signuptest.jsx [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+        email: 'test@test.com',
+        emailVerified: true,
+        getIdToken: () => new Promise(((resolve) => {
+          setTimeout(() => {
+            resolve(1234);
+          }, 0);
+        })),
+      },
+    }));
+  it('Render Signup without any eror', () => { // eslint-disable-line no-undef
```
Comment (khelif96) [inline website/src/views/Signup/Signuptest.jsx]: Comments

**Comment by khelif96** [comment]
Is this still needed?

**Comment by LiuYuHao** [comment]
Yes! The only two problems are in sign in and sigh out. The mock history push. I am sorry didn't do anything during thanksgiving break

**Comment by LiuYuHao** [comment]
Also this snapshot test is not the test you guys are going to write since one snapshot test is covered everything. I will make a new pr about shallow. We can merge this pr first and you guys can make test cases beside signin page and signup page until I will make a pr to fix them.

**Comment by LiuYuHao** [comment]
Except the SignUp everything is done. I will make pr about shallow(but the problem is the snapshot case is in this branch, we can merge this or copy the test file which u are wroking on shallow from this pr)

**Pull request closed without merging**